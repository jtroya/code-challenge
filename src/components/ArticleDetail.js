import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ArticleDetail extends Component {
  componentDidMount() {
    this.props.getArticle(this.props.id);
  }

  handleButtonBack = () => {
    this.props.history.push('/');
  }

  render() {
    const { author, content, title, tags, published } = this.props.selectedItem;
    const backButton = <div><button onClick={this.handleButtonBack}>Volver</button></div>;
    const isPublished = published ? 'Yes' : 'No';
    const TagList = ({ items }) =>
      <ul>
        { items ? items.map((item, index) => (
          <li key={index}>
            <span>{ item }</span>
          </li>))
          : null
        }
      </ul>;
    return (
      <div className="container-article">
          <div className="box-info">
            { backButton }
            <h2>{ title }</h2>
            <h3>{ author }</h3>
            <p>{ content }</p>
            <h4>Tags:</h4>
            <TagList items={tags} />
            <p>Published: { isPublished }</p>
            { backButton }
          </div>
      </div>
    );
  }
}

ArticleDetail.propTypes = {
  getArticle: PropTypes.func,
  history: PropTypes.shape({ push: PropTypes.func }),
  id: PropTypes.string,
  selectedItem: PropTypes.shape(
    {
      id: PropTypes.string,
      author: PropTypes.string,
      content: PropTypes.string,
      tags: PropTypes.array,
      title: PropTypes.string,
      published: PropTypes.bool,
    }),
};

export default ArticleDetail;
