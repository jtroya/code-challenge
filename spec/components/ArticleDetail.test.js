import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ArticleDetail from '../../src/components/ArticleDetail';

describe('ArticleDetail', () => {
  let wrapper = null;
  const getArticle = jest.fn();
  const idSelected = 'ABC123';
  const fakeSelectedItem = {
    id: 'ABC123',
    author: 'my author',
    content: 'the content',
    title: 'the title',
  };

  beforeEach(() => {
    wrapper = shallow(
      <ArticleDetail
        id={idSelected}
        getArticle={getArticle}
        selectedItem={fakeSelectedItem}
      />);
  });

  it('should render the component', () => {
    expect(wrapper.find('ArticleDetail')).toBeTruthy();
  });

  it('should matches the snapshot', () => {
    const tree = shallow(
      <ArticleDetail
        id={idSelected}
        getArticle={getArticle}
        selectedItem={fakeSelectedItem}
      />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
