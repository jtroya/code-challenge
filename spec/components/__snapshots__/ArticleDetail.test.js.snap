exports[`ArticleDetail should matches the snapshot 1`] = `
<div
  className="container-article">
  <div
    className="box-info">
    <div>
      <button
        onClick={[Function]}>
        Volver
      </button>
    </div>
    <h2>
      the title
    </h2>
    <h3>
      my author
    </h3>
    <p>
      the content
    </p>
    <h4>
      Tags:
    </h4>
    <TagList />
    <p>
      Publised: 
      No
    </p>
    <div>
      <button
        onClick={[Function]}>
        Volver
      </button>
    </div>
  </div>
</div>
`;
